tests/unit/tests/eviction/eviction.c/eviction.c

```diff
@@ -0,0 +1,290 @@
+	struct ocf_user_part upart[OCF_IO_CLASS_MAX];
+	uint32_t overflow[OCF_IO_CLASS_MAX];
+	uint32_t evictable[OCF_IO_CLASS_MAX];
+};
+
+bool __wrap_ocf_eviction_can_evict(ocf_cache_t cache)
+{
+	return true;
+}
+
+uint32_t __wrap_ocf_part_overflow_size(struct ocf_cache *cache,
+		struct ocf_user_part *part)
+{
+	struct test_cache* tcache = cache;
+
+	return tcache->overflow[part->id];
+}
+
+uint32_t __wrap_ocf_evict_calculate(ocf_cache_t cache,
+		struct ocf_user_part *part, uint32_t to_evict, bool roundup)
+{
+	struct test_cache* tcache = cache;
+
+	return min(tcache->evictable[part->id], to_evict);
+}
+
+uint32_t __wrap_ocf_eviction_need_space(struct ocf_cache *cache,
+	ocf_queue_t io_queue, struct ocf_user_part *part,
+	uint32_t clines)
+{
+	struct test_cache *tcache = (struct test_cache *)cache;
+	unsigned overflown_consumed = min(clines, tcache->overflow[part->id]);
+
+	tcache->overflow[part->id] -= overflown_consumed;
+	tcache->evictable[part->id] -= clines;
+
+	check_expected(part);
+	check_expected(clines);
+	function_called();
+	return mock();
+}
+
+int __wrap_ocf_log_raw(ocf_logger_t logger, ocf_logger_lvl_t lvl,
+		const char *fmt, ...)
+{
+}
+
+int __wrap_ocf_log_stack_trace_raw(ocf_logger_t logger)
+{
+	return 0;
+}
+
+ocf_ctx_t __wrap_ocf_cache_get_ctx(ocf_cache_t cache)
+{
+	return NULL;
+}
+
+
+bool ocf_cache_is_device_attached(ocf_cache_t cache)
+{
+	return true;
+}
+
+
+/* FIXME: copy-pasted from OCF */
+int ocf_part_lst_cmp_valid(struct ocf_cache *cache,
+		struct ocf_lst_entry *e1, struct ocf_lst_entry *e2)
+{
+	struct ocf_user_part *p1 = container_of(e1, struct ocf_user_part,
+			lst_valid);
+	struct ocf_user_part *p2 = container_of(e2, struct ocf_user_part,
+			lst_valid);
+	size_t p1_size = ocf_cache_is_device_attached(cache) ?
+				p1->runtime->curr_size : 0;
+	size_t p2_size = ocf_cache_is_device_attached(cache) ?
+				p2->runtime->curr_size : 0;
+
+	int v1 = p1->config->priority;
+	int v2 = p2->config->priority;
+
+	/*
+	 * If partition is invalid the priority depends on current size:
+	 * 1. Partition is empty - move to the end of list
+	 * 2. Partition is not empty  - move to the beginning of the list. This
+	 * partition will be evicted first
+	 */
+
+	if (p1->config->priority == OCF_IO_CLASS_PRIO_PINNED)
+		p1->config->flags.eviction = false;
+	else
+		p1->config->flags.eviction = true;
+
+	if (p2->config->priority == OCF_IO_CLASS_PRIO_PINNED)
+		p2->config->flags.eviction = false;
+	else
+		p2->config->flags.eviction = true;
+
+	if (!p1->config->flags.valid) {
+		if (p1_size) {
+			v1 = SHRT_MAX;
+			p1->config->flags.eviction = true;
+		} else {
+			v1 = SHRT_MIN;
+			p1->config->flags.eviction = false;
+		}
+	}
+
+	if (!p2->config->flags.valid) {
+		if (p2_size) {
+			v2 = SHRT_MAX;
+			p2->config->flags.eviction = true;
+		} else {
+			v2 = SHRT_MIN;
+			p2->config->flags.eviction = false;
+		}
+	}
+
+	if (v1 == v2) {
+		v1 = p1 - cache->user_parts;
+		v2 = p2 - cache->user_parts;
+	}
+
+	return v2 - v1;
+}
```

Comment:
Instead of copy pasting you can add `ocf_part_lst_cmp_valid` to <functions_to_leave> list in test's header

Reply:
It won't cut it. I think it's because ocf_part_lst_cmp_valid is defined in another file.